kpconv_torch/train.py

```diff
@@ -80,174 +68,120 @@ def train(
     print("Data Preparation")
     print("****************")
 
-    # Initialize configuration class
-    if dataset == "ModelNet40":
-        config = ModelNet40Config()
-    if dataset == "NPM3D":
-        config = NPM3DConfig()
-    if dataset == "S3DIS":
-        config = S3DISConfig()
-    if dataset == "SemanticKitti":
-        config = SemanticKittiConfig()
-    elif dataset == "Toronto3D":
-        config = Toronto3DConfig()
-
-    config.set_output_dir(output_dir)
-    config.set_chosen_log(chosen_log)
-
-    # overload training process length and checkpoint saving frequency
-    if max_epoch is not None:
-        config.max_epoch = max_epoch
-    if checkpoint_gap is not None:
-        config.checkpoint_gap = checkpoint_gap
-    if epoch_steps is not None:
-        config.epoch_steps = epoch_steps
-    if validation_size is not None:
-        config.validation_size = validation_size
-
     train_save_path = get_train_save_path(output_dir, chosen_log)
-
-    if chosen_log:
-        config.load(train_save_path)
-        if config.dataset != dataset:
-            raise ValueError(
-                f"Config dataset ({config.dataset}) "
-                f"does not match provided dataset ({dataset})."
-            )
-        config.chosen_log = None
-
-    # Get path from argument if given
-    if len(sys.argv) > 1:
-        config.chosen_log = sys.argv[1]
+    if chosen_log is None:
+        config_file_path = configfile
+    else:
+        config_file_path = Path(train_save_path / "config.yml")
```

Comment:
It is tested [here](https://github.com/delhomer/kpconv-torch/pull/57/files#diff-e34ada1abf156e2b36d35873bd3ab6976fd15b794349d0a62163ae54d972d852R129), in the CLI that there is at least `chosen_log` or `configfile` that is defined. So, it should not be possible at this step of the code. If `chosen_log` is defined, we take the configuration file from inside the folder, otherwise, we take the one defined by the CLI.